python/packages/kagent-adk/tests/unittests/models/test_tls_integration.py

```diff
@@ -0,0 +1,422 @@
+    with caplog.at_level(logging.INFO):
+        if ca_cert_path == "fake_path":
+            ca_cert_path = temp_cert_file
+
+        if expected_mode == "disabled":
+            ctx = create_ssl_context(
+                verify_disabled=verify_disabled,
+                ca_cert_path=ca_cert_path,
+                use_system_cas=use_system_cas,
+            )
+            assert ctx is False
+            assert "TLS Mode: Disabled" in caplog.text
+        else:
+            with mock.patch("ssl.create_default_context") as mock_default_ctx:
+                with mock.patch.object(ssl.SSLContext, "load_verify_locations"):
+                    mock_ctx = mock.MagicMock()
+                    if use_system_cas:
+                        mock_default_ctx.return_value = mock_ctx
+                    else:
+                        # For custom_only mode, mock SSLContext constructor
+                        with mock.patch("ssl.SSLContext") as mock_ssl_ctx:
+                            mock_ssl_ctx.return_value = mock_ctx
+                            ctx = create_ssl_context(
+                                verify_disabled=verify_disabled,
+                                ca_cert_path=ca_cert_path,
+                                use_system_cas=use_system_cas,
+                            )
+                            assert ctx is mock_ctx
+                        return
+
+                    ctx = create_ssl_context(
+                        verify_disabled=verify_disabled,
+                        ca_cert_path=ca_cert_path,
+                        use_system_cas=use_system_cas,
+                    )
+                    assert ctx is mock_ctx
+
+            # Verify appropriate mode is logged
+            if expected_mode == "system_only":
+                assert "System CAs only" in caplog.text
+            elif expected_mode == "custom_and_system":
+                assert "Custom CA + System CAs" in caplog.text
+
+
+# ============================================================================
+# Integration Test 6: SSL Error Troubleshooting Message
+# ============================================================================
+
+
+def test_e2e_ssl_error_troubleshooting_message(temp_cert_file):
+    """Test that SSL errors generate helpful troubleshooting messages."""
+    error = ssl.SSLError("certificate verify failed")
+
+    message = get_ssl_troubleshooting_message(
+        error=error,
+        ca_cert_path=temp_cert_file,
+        server_url="litellm.internal.corp:8080",
+    )
+
+    # Verify troubleshooting message contains actionable guidance
+    assert "SSL/TLS Connection Error" in message
+    assert "kubectl exec" in message
+    assert "openssl x509" in message
+    assert "openssl s_client" in message
+    assert temp_cert_file in message
+    assert "litellm.internal.corp:8080" in message
+    assert "https://docs.kagent.dev/troubleshooting/ssl-errors" in message
```

Comment:
Could we remove or update this from the logging if there's a PR that has relevant troubleshooting? I don't think the page currently exists & the kagent docs page is `https://kagent.dev/docs`